pkg/git/gogit/checkout_test.go

```diff
@@ -359,6 +371,359 @@ func TestCheckoutTagSemVer_Checkout(t *testing.T) {
 	}
 }
 
+// Test_KeyTypes assures support for the different types of keys
+// for SSH Authentication supported by Flux.
+func Test_KeyTypes(t *testing.T) {
+	tests := []struct {
+		name       string
+		keyType    ssh.KeyPairType
+		authorized bool
+		wantErr    string
+	}{
+		{name: "RSA 4096", keyType: ssh.RSA_4096, authorized: true},
+		{name: "ECDSA P256", keyType: ssh.ECDSA_P256, authorized: true},
+		{name: "ECDSA P384", keyType: ssh.ECDSA_P384, authorized: true},
+		{name: "ECDSA P521", keyType: ssh.ECDSA_P521, authorized: true},
+		{name: "ED25519", keyType: ssh.ED25519, authorized: true},
+		{name: "unauthorized key", keyType: ssh.RSA_4096, wantErr: "unable to authenticate, attempted methods [none publickey], no supported methods remain"},
+	}
+
+	serverRootDir := t.TempDir()
+	server := gittestserver.NewGitServer(serverRootDir)
+
+	// Auth needs to be called, for authentication to be enabled.
+	server.Auth("", "")
```

Comment:
Looks like this API element could be improved in `fluxcd/pkg` to save ourselves from having to make this comment anywhere else.

Reply:
Agreed - I will make sure we do that. :+1: